internal/providers/terraform/parser.go

```diff
@@ -493,7 +493,13 @@ func (p *Parser) parseReferences(resData map[string]*schema.ResourceData, conf g
 func (p *Parser) parseConfReferences(resData map[string]*schema.ResourceData, conf gjson.Result, d *schema.ResourceData, attr string, registryMap *ResourceRegistryMap) bool {
 	// Check if there's a reference in the conf
 	resConf := getConfJSON(conf, d.Address)
-	refResults := resConf.Get("expressions").Get(attr).Get("references").Array()
+	exps := resConf.Get("expressions").Get(attr)
+	lookupStr := "references"
+	if exps.IsArray() {
+		lookupStr = "#.references"
+	}
```

Comment:
This fixes issue where nested blocks which contained references could not be parsed. Using the following expression JSON

```json
  "expressions": {
      "subnet_mapping": [
        {
          "allocation_id": {
            "references": [
              "aws_eip.eip4.id",
              "aws_eip.eip4"
            ]
          },
        }
      ]
    }
```

If a reference of `subnet_mapping.#.allocation_id` would fail to return the references.